dts/bindings/memory-controllers/infineon,xmc4xxx-ebu.yaml

```diff
@@ -0,0 +1,216 @@
+              bus-write-timing = <XMC4XXX_EBU_SET_BUSWAP(15, 7, 4, 0, 0, 0, 15, 15, 15)>;
+      };
+      status = "okay";
+  };
+
+  The memory region is accessible by defining the node
+
+  sdram1: sdram@60000000 {
+      compatible = "zephyr,memory-region", "mmio-sram";
+      device_type = "memory";
+      reg = <0x60000000 DT_SIZE_M(8)>;
+      zephyr,memory-region = "SDRAM1";
+  };
+
+  The following properties define the EBU registers outlined in the XMC4XXX reference
+  manual. A large number of configurations are exposed and it is recommended
+  to carefully review the XMC4XXX manual to choose the proper settings for your
+  target device.
+
+  Each of the properties has a macro to simplify setting the values. The bitflags
+  use the same names as defined in the reference manual.
+
+compatible: "infineon,xmc4xxx-ebu"
+
+include: [base.yaml, pinctrl-device.yaml]
+
+properties:
+  reg:
+    required: true
+
+  "#address-cells":
+    required: true
+    const: 1
+
+  "#size-cells":
+    required: true
+    const: 0
+
+  clk-config:
+    type: int
+    required: true
+    description: |
+      Configures Clock Control (CLC) register. Controls which clock is used for the EBU and
+      the clock divider. Use XMC4XXX_EBU_SET_CLC(sync, div2, ebudiv).
+
+  modes-config:
+    type: int
+    required: true
+    description: |
+      Sets Modes Configuration (MODCON) register. This mainly sets the arbitration parameters.
+      Also it is used to configure whether address latch enable (ALE) is active high or low and whether SDRAM
+      pins CKE, SDCLKO, CAS, and RAS are in tri-state when the EBU does not own the external bus.
+      Use macro XMC4XXX_EBU_SET_MODCON(sdtri, extlock, arbsync, arbmode, timeoutc, ale).
+
+  gpio-control-config:
+    type: int
+    required: true
+    description: |
+      Configures USERCON register. This register allows the user to free unused gpio pins.
+      Use macro XMC4XXX_EBU_SET_USERCON(addio, advio).
+
+  sdram-control:
+    type: int
+    required: false
+    description: |
+      Configures SDRMCON register.
+      Use macro XMC4XXX_EBU_SET_SDRMCON(sdcmsel, pwr_mode, clkdis, crce, bankm,
+      rowm, crc, crcd, awidth, crp, crsc, crfsh, cras).
+
+  sdram-operation-mode:
+    type: int
+    required: false
+    description: |
+      Configures SDRMOD register.
+      Use macro XMC4XXX_EBU_SDRMOD(xba, xopm, coldstart, opmode, caslat, burstl).
+
+  sdram-refresh-control:
+    type: int
+    required: false
+    description: |
+      Configures SDRMREF register.
+      Use macro XMC4XXX_EBU_SDRMREF(res_dly, arfsh, selfrex_dly, erfshc, autoselfr,
+      selfrex, refreshr, refreshc).
+
+child-binding:
+  description: |
+    Child device nodes are representing devices connected to the EBU bus.
+
+  properties:
+    reg:
+      type: int
+      required: true
+      description: Selects the region to configure. All the following properties apply to this region.
+
+    device-type:
```

Comment:
please come up with a different property name.  `device-type` is a standard devicetree property.

Reply:
I added a infineon prefix to the child bindings